components/script/dom/htmlimageelement.rs

```diff
@@ -1159,6 +1170,43 @@ impl HTMLImageElement {
         }
     }
 
+    // Step 2 for <https://html.spec.whatwg.org/multipage/#dom-img-decode>
+    fn react_to_decode_image_sync_steps(&self) {
+        let document = document_from_node(self);
+        let elem = self.upcast::<Element>();
+        // Step 2.1 of <https://html.spec.whatwg.org/multipage/#dom-img-decode>
+        if !document.is_fully_active() ||
+            matches!(self.current_request.borrow().state, State::Broken)
+        {
+            self.reject_image_decode_promises();
+        } else if matches!(
+            self.current_request.borrow().state,
+            State::CompletelyAvailable
+        ) {
+            self.resolve_image_decode_promises();
+        } else if !elem.has_attribute(&local_name!("src")) &&
+            !elem.has_attribute(&local_name!("srcset"))
+        {
+            self.reject_image_decode_promises();
+        }
```

Comment:
@gterzian, I need to add this condition(src, srcset) to make other tests pass, but I am not sure if this is the correct place. Also, two remaining SVG tests are failing. Does Servo Image support SVG?

Reply:
I will remove this one because does not align with spec